client/src/boot/transforms.js

```diff
@@ -0,0 +1,27 @@
+import Injector from 'lib/Injector';
+import classnames from 'classnames';
+
+export default () => {
+  Injector.transform(
+    'form-action-changed',
+    (updater) => {
+      updater.form.alterSchema('*', form => {
+        form.mutateField('action_save', (field) => {
+          const isPristine = form.isPristine();
+          const classes = classnames({
+            [field.data.pristineClass]: isPristine,
+            [field.data.dirtyClass]: !isPristine,
+          });
+          return {
+            ...field,
+            title: isPristine ? field.data.pristineTitle : field.data.dirtyTitle,
```

Comment:
What if pristineTitle and dirtyTitle aren't set? FormField.php doesn't set these by default either.